Patch extracted from a set of patches to cross-compile ffmeg and dependencies for mingw-w64.
https://github.com/Warblefly/MultimediaTools-mingw-w64/commit/01e30a17637b67afd65c1e24302f8daf94b19b32#diff-503bd595ad2a49840cc56acdb3432487a20a0acb7981ce83be51fe311216de8f

diff --git a/include/CMakeLists.txt b/include/CMakeLists.txt
index 99a69fd..e5ba8a7 100644
--- a/include/CMakeLists.txt
+++ b/include/CMakeLists.txt
@@ -35,8 +35,6 @@ endif(VA_LIST_IS_ARRAY)
 INCLUDE(TestBigEndian)
 TEST_BIG_ENDIAN(WORDS_BIGENDIAN)
 
-TRY_RUN(BITFIELDS_HTOL TEST_DUMMY ${CMAKE_BINARY_DIR} ${CMAKE_CURRENT_SOURCE_DIR}/test_BITFIELDS_HTOL.c)
-
 INCLUDE(CheckIncludeFiles)
 
 #SET(CMAKE_REQUIRED_INCLUDES "/usr/include;/usr/local/include")
diff --git a/include/xconfig.h.in b/include/xconfig.h.in
index c130600..476c00b 100644
--- a/include/xconfig.h.in
+++ b/include/xconfig.h.in
@@ -233,7 +233,11 @@
 /* If using network byte order             */
 #cmakedefine WORDS_BIGENDIAN
 /* If high bits come first in structures   */
-#cmakedefine BITFIELDS_HTOL
+#ifdef WORDS_BIGENDIAN
+#define BITFIELDS_HTOL
+#else
+#define BITFIELDS_LTOH
+#endif
 #define	HAVE_C_BIGENDIAN	/* Flag that WORDS_BIGENDIAN test was done */
 #define	HAVE_C_BITFIELDS	/* Flag that BITFIELDS_HTOL test was done  */
